src/Sema.zig

```diff
@@ -5785,6 +5785,14 @@ fn zirIntToPtr(sema: *Sema, block: *Scope.Block, inst: Zir.Inst.Index) InnerErro
         });
     }
     try sema.requireRuntimeBlock(block, src);
+    if (block.wantSafety()) {
+        const zero = try sema.mod.constInst(sema.arena, src, .{
+            .ty = Type.initTag(.u64),
+            .val = Value.initTag(.zero),
+        });
+        const is_non_zero = try block.addBinOp(src, Type.initTag(.bool), .cmp_neq, operand_coerced, zero);
```

Comment:
This is wrong as some pointer types do accept zero as value. Check out `ir_analyze_int_to_ptr`, the pointer alignment must be validated too.

Reply:
Right, thanks fixed.